notify/pagerduty/pagerduty.go

```diff
@@ -107,6 +110,37 @@ type pagerDutyPayload struct {
 	CustomDetails map[string]string `json:"custom_details,omitempty"`
 }
 
+func (n *Notifier) encodeMessage(msg *pagerDutyMessage) (bytes.Buffer, error) {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
+		return buf, errors.Wrap(err, "failed to encode PagerDuty message")
+	}
+
+	if buf.Len() >= MaxEventSize {
+		truncatedMsg := fmt.Sprintf("Custom details have been removed because the original event exceeds the maximum size of %s", units.MetricBytes(MaxEventSize).String())
```

Comment:
Can this be a package-level variable?

Reply:
@roidelapluie  If I add something like
```
const TruncatedCustomDetailsMsg string = fmt.Sprintf("Custom details have been removed because the original event exceeds the maximum size of %s", units.MetricBytes(MaxEventSize).String())
```
I get a compile error because the expression is not a constant. 
I can have something like
```
const TruncatedCustomDetailsMsg string = "Custom details have been removed because the original event exceeds the maximum size of %s"
```
or have the size limit hardcoded in the message
```
const TruncatedCustomDetailsMsg string = "Custom details have been removed because the original event exceeds the maximum size of 512KB"
```
Which do you prefer?

Reply:
var TruncatedCustomDetailsMsg string = fmt.Sprintf("Custom details have been removed because the original event exceeds the maximum size of %s", units.MetricBytes(MaxEventSize).String())

Reply:
My point is that it is maybe not needed to compute it upon each request

Reply:
Ah! I thought package level variables can only be `const`
Sorry, I am new to golang :)
I will make the change.

Reply:
Also it can be unexported

Reply:
It is important to have it out of the function because it is a recursively function. We need to be sure it is the same between iterations. While it was true I'm your implementation, future code changes might have created an infinite loop.